contracts/Splitter.sol

```diff
@@ -7,34 +7,29 @@ contract Splitter {
     
     address payable[] internal recipients = [0x20601F1Ddb44E28b1511EdBD316A368D80116408, //"Bob";
                                             0x24F1500890505ceD8534502ab403F20ce99feea0]; //"Carol";
+    uint InitialValue = 100 ether;
 
     event Deposit(uint date,
                   address indexed _from,
                   address indexed _to,
                   uint amount);
 
     function deposit() external payable{
-        doDeposit(msg.sender, msg.value);
-    }
 
-    function doDeposit(address _sender, uint _value) public payable {
-        if(_sender != sender){
+        if(msg.value % 2 != 0) {
```

Comment:
You're making the sender to a check for even values which could be burdensome to clients, some of which might be external contracts that compute interest, dividend, royalty, and might not be able to deal with this. 

Have you considered returning the change?

Reply:
I have to admit that never occurred to me! I'll try that.